src/domains/wallet/pages/ImportWallet/ImportWallet.tsx

```diff
@@ -36,131 +40,102 @@ export const ImportWallet = () => {
 	const { selectedNetworkIds, setValue } = useDashboardConfig({ profile: activeProfile });
 
 	const { t } = useTranslation();
+	const { importWalletByType } = useWalletImport({ profile: activeProfile });
+	const { syncAll } = useWalletSync({ profile: activeProfile, env });
 
 	const form = useForm({ mode: "onChange", defaultValues: { type: "mnemonic" } });
-	const { formState, register } = form;
+	const { formState, register, watch, handleSubmit, unregister } = form;
 	const { isSubmitting, isValid } = formState;
+	const { value, encryptionPassword, confirmEncryptionPassword } = watch();
 
 	const nameMaxLength = 42;
 
 	useEffect(() => {
 		register({ name: "type", type: "custom" });
 	}, [register]);
 
-	const handleBack = () => {
-		if (activeTab === 1) {
-			return history.push(`/profiles/${activeProfile.id()}/dashboard`);
+	useEffect(() => {
```

Comment:
this `useEffect` can probably be avoided by using `value` directly instead of mirroring it to `walletGenerationInput` and using that.

Reply:
Yes, the reason that I did it, was because value is lost after the 2nd step renders, and with this I cache it so to use it on the import wallet step. Tried using `shouldUnregister:false` in the form but it breaks many parts of the existing implementation around the `isValid` state of the form(improperly disabling buttons, validates passed steps etc). So it seemed that we need the validation for each step separately `shouldUnregister: true` but we only needed the `value` to import the wallet.